hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/LdapGroupsMapping.java

```diff
@@ -437,8 +443,14 @@ Set<String> lookupGroup(SearchResult result, DirContext c,
     Set<String> groupDNs = new HashSet<>();
 
     NamingEnumeration<SearchResult> groupResults;
-    // perform the second LDAP query
-    if (isPosix) {
+
+    String[] resolved = resolveCustomGroupFilterArgs(result);
+    // If custom group filter argument is supplied, use that!!!
+    if (resolved != null) {
```

Comment:
Is it not the case that this condition is a subset of the non-posix condition check? I think that we should only be looking for customGroupFilterArgs if it is non-posix not before the check of isPosix. Am I missing something that makes this the appropriate place for this?

Reply:
One of the filter that we checked internally had the object class as posixGroup and it can still satisfy the isPosix condition, but if some user has explicitly configured that replace my {0} with say X and {1} with say Y, so we should do that irrespective of whether the filter anywhere contains posixGroup or not.
Else if he has this `posixGroup` and stuff in the group filter, we can't use custom filters at all and my use case will break.

Can we justify like if someone configured it use custom filter then we use that because he explicitly asked us to do so. If he doesn't then we do our normal logic and figure out if it is posix or non posix.

By default it will always be turned off, because there is no default value set, so won't(or I should say shouldn't) bother any existing use case.

Reply:
I see. I was misinterpreting what isPOSIX was representing. This is resolved.